lectures/12-plonk.tex

```diff
@@ -0,0 +1,313 @@
+\documentclass[../lecture-notes.tex]{subfiles}
+\usepackage[T1]{fontenc}
+\usepackage{lmodern}
+\usepackage{xcolor}
+\usepackage{fix-cm}
+\usepackage{amssymb}
+\pgfplotsset{compat=1.18}
+
+\begin{document}
+\subsection{Preliminaries}
+Before delving into Plonk, let's first consider a few useful proving system gadgets.
+
+Let $\Omega$ be some multiplicative cyclic subgroup (recall the \ref{section:mult-group-finite-field}) of $\mathbb{F}_p$ of size $k$. 
+
+Let $f \in \mathbb{F}_p^{(\leq d)}[X] \qquad (d \geq k)$ \begin{tabular}{|c|}\hline Verifier has $f$ \\ \hline \end{tabular} 
+
+Now we need to construct efficient Poly-IOPs for two tasks described in below sections.
+
+\subsubsection{ZeroTest}
+
+Prove that $f$ is identically zero on $\Omega$.
+
+\begin{figure}[H]
+  \centering
+  \begin{tikzpicture}
+      \node[inner sep=0pt, align=center] (prover) {\includegraphics[width=1.25cm]{lectures/images/common/prover.png}\\Prover $\mathcal{P}$}; 
+      \node[inner sep=0pt, align=center, right=5cm of prover] (verifier) {\includegraphics[width=1.25cm]{lectures/images/common/verifier.png}\\Verifier $\mathcal{V}$}; 
+
+      \draw [dashed,line width=0.3mm] ([yshift=-0.5cm]prover.south) -- ([yshift=-10cm]prover.south);
+      \draw [dashed,line width=0.3mm] ([yshift=-0.5cm]verifier.south) -- ([yshift=-10cm]verifier.south);
+
+      \node[align=center,fill=white!5,thick,below=0.5cm of prover](prover-compute){
+      \noindent\rule{3.5cm}{0.8pt}\\
+      Compute: \( q(X) \leftarrow \frac{f(X)}{Z_{\Omega}(X)} \) \\ \( q \in \mathbb{F}_p^{(\leq d)}[X] \) \\
+      \noindent\rule{3.5cm}{0.8pt}};
+
+      \draw[-{Stealth[length=3mm]},line width=0.4mm] ([yshift=-3cm]prover.south) coordinate (l2)--(l2-|verifier) node[midway, above=0mm]{Send \(com(q)\)};
+
+      \node[align=center,fill=white!5,thick,below=3.5cm of verifier](verifier-random){
+      \noindent\rule{3.5cm}{0.8pt}\\
+      Generate random \(r \xleftarrow{R}\mathbb{F}_p^{(\leq d)} \) \\
+      \noindent\rule{3.5cm}{0.8pt}};
+
+      \draw[-{Stealth[length=3mm]},line width=0.4mm] ([yshift=-5.5cm]verifier.south) coordinate (l2)--(l2-|prover) node[midway, above=0mm]{Query \(q\) and \(f\) at $r$};
+
+      \node[align=center,fill=white!5,thick,below=6.0cm of prover](prover-query){
+      \noindent\rule{3.5cm}{0.8pt}\\
+      Compute: $q(r), f(r)$ \\
+      \noindent\rule{3.5cm}{0.8pt}};
+
+      \draw[-{Stealth[length=3mm]},line width=0.4mm] ([yshift=-8cm]prover.south) coordinate (l2)--(l2-|verifier) node[midway, above=0mm]{Send \(q(r), f(r)\)};
+
+      \node[align=center,fill=white!5,thick,below=8.5cm of verifier](verifier-accept){
+      \noindent\rule{3.5cm}{0.8pt}\\
+      Accept if $f(r) = q(r) \cdot Z_{\Omega}(r)$ \\
```

Comment:
There is no explanation of what $Z_{\Omega}(X)$ is (we did consider it in the previous lectures, but at least specify that it is $Z_{\Omega}(X) = \prod_{\omega \in \Omega}(X-\omega)$ and if the group is cyclic, this reduces to 

$$
Z_{\Omega}(X) = X^{|\Omega|} - 1
$$

Reply:
Also, what is the purpose for the verifier of having the commitment to $f$? Specify where this fact is used